### PR TITLE
fix too dark or bright primary element color

### PR DESCRIPTION
**Pull request by szaimen**
To reproduce: choose a very dark primary color and select the dark mode.

| Before | After |
|---|---|
| image | image |

### CODE REVIEW

**Review by nimishavijay — COMMENTED** [review]
Looks good in the screenshot 👍 as long as it is not affected in light mode it is good to go

File: apps/theming/lib/IconBuilder.php [hunk 1]
```diff
@@ -234,7 +234,13 @@ public function colorSvg($app, $image) {
 		}
 		$svg = file_get_contents($imageFile);
 		if ($svg !== false && $svg !== "") {
-			$color = $this->util->elementColor($this->themingDefaults->getColorPrimary());
+			$primaryColor = $this->themingDefaults->getColorPrimary();
+			$luminance = $this->util->calculateLuminance($primaryColor);
+			$isBrightColor = false;
+			if ($luminance > 0.8) {
```
Comment (nickvergessen) [inline apps/theming/lib/IconBuilder.php:240]: and cut off into the other direction for bright mode?

Comment (szaimen) [reply apps/theming/lib/IconBuilder.php:240]: The cut-off for dark-mode is already handled in elementColor so both will be fine. I verified this already

**Comment by szaimen** [comment]
Can someone please restart drone for me? Thanks a lot!

File: apps/theming/lib/IconBuilder.php (hunk 1 above)
Comment (szaimen) [reply apps/theming/lib/IconBuilder.php:240]: good point

File: apps/theming/tests/UtilTest.php [hunk 2]
```diff
@@ -107,17 +107,17 @@ public function testInvertTextColorEmpty() {
 
 	public function testElementColorDefault() {
 		$elementColor = $this->util->elementColor("#000000");
```
Comment (nickvergessen) [inline apps/theming/tests/UtilTest.php:109]: also test the function with true as second parameter with `#000000` and `#ffffff`?

Comment (szaimen) [reply apps/theming/tests/UtilTest.php:109]: done

**Review by nickvergessen — APPROVED** [review]

**Review by susnux — APPROVED** [review]
Works, very nice!

**Review by jancborchardt — APPROVED** [review]
Looks good! :)

**Comment by szaimen** [comment]
CI failure unrelated

**Comment by szaimen** [comment]
I wonder should we backport this or rather not?

**Pull request merged**